compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -906,6 +906,7 @@ object Parsers {
       var braces = 0
       while (true) {
         val token = lookahead.token
+        if (query != LARROW && token == XMLSTART) return false
```

Comment:
This fixes the following case,

```scala
  val casePatMatch = for (case t @ <foo>FooBar</foo> <- Seq(xml))
   yield t
```

but it still fails with the following case.

```scala
  val casePatMatchWithCond = for (case t @ <foo>FooBar</foo> <- Seq(xml))
   yield t
```

I need to investigate routines more.

Reply:
Ah, I made a mistake.

```diff
- val casePatMatchWithCond = for (case t @ <foo>FooBar</foo> if true <- Seq(xml))
+ val casePatMatchWithCond = for (case t @ <foo>FooBar</foo><- Seq(xml) if true )
   yield t
```